lib/analyzer/package-managers/apt.ts

```diff
@@ -43,25 +44,55 @@ export function analyzeDistroless(
 
 function parseDpkgFile(text: string): AnalyzedPackageWithVersion[] {
   const pkgs: AnalyzedPackageWithVersion[] = [];
-  let curPkg: any = null;
+  let curPkg: AnalyzedPackageWithVersion | null = null;
   for (const line of text.split("\n")) {
-    curPkg = parseDpkgLine(line, curPkg, pkgs);
+    curPkg = parseDpkgLine(line, curPkg!, pkgs);
+    if (curPkg) {
+      curPkg.Purl = purl(curPkg);
+    }
   }
   return pkgs;
 }
 
+export function purl(curPkg: AnalyzedPackageWithVersion): string | undefined {
+  if (!curPkg.Name || !curPkg.Version) {
+    return undefined;
+  }
+
+  let qualifiers: { upstream: string } | undefined;
+  if (curPkg.Source && curPkg.SourceVersion) {
+    qualifiers = {
+      upstream: encodeURIComponent(`${curPkg.Source}@${curPkg.SourceVersion}`),
+    };
+  } else if (curPkg.Source) {
+    qualifiers = {
+      upstream: encodeURIComponent(curPkg.Source),
+    };
+  }
+
+  return new PackageURL(
+    "deb",
+    "",
+    curPkg.Source ? `${curPkg.Source}/${curPkg.Name}` : curPkg.Name,
```

Comment:
Again I think this should *only* be the name (and the validation ensures that as well right now).

But maybe I'm missing something, let me know if you want to pair on this.